### PR TITLE
OpenEXR: Url fix, package type fix, and update

### PR DESCRIPTION
**Pull request by dorton21**

### CODE REVIEW

File: var/spack/repos/builtin/packages/openexr/package.py [hunk 1]
```diff
@@ -6,14 +6,14 @@
 from spack import *
 
 
-class Openexr(AutotoolsPackage):
+class Openexr(CMakePackage):
     """OpenEXR Graphics Tools (high dynamic-range image file format)"""
 
     homepage = "http://www.openexr.com/"
-    url = "https://github.com/openexr/openexr/releases/download/v2.3.0/openexr-2.3.0.tar.gz"
+    url      = "https://github.com/AcademySoftwareFoundation/openexr/archive/v2.5.3.tar.gz"
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openexr/package.py:13]: spacing after variable name seems wrong (s/b single space)

File: var/spack/repos/builtin/packages/openexr/package.py [hunk 2]
```diff
@@ -6,14 +6,14 @@
...
 
-    # New versions should come from github now
-    version('2.3.0', sha256='fd6cb3a87f8c1a233be17b94c74799e6241d50fc5efd4df75c7a4b9cf4e25ea6')
+    # New Version Should Come From GitHub
+    version('2.5.3', sha256='6a6525e6e3907715c6a55887716d7e42d09b54d2457323fcee35a0376960bebf')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openexr/package.py:15]: unless there's a good reason to remove 2.3.0, 2.5.3 should be added not replace it

File: var/spack/repos/builtin/packages/openexr/package.py [hunk 3]
```diff
@@ -32,9 +32,8 @@ class Openexr(AutotoolsPackage):
     version('1.3.2', sha256='fa08ad904bf89e2968078d25d1d9817f5bc17f372d1bafabf82e8f08ca2adc20',
             url="http://download.savannah.nongnu.org/releases/openexr/openexr-1.3.2.tar.gz")
 
-    variant('debug', default=False,
-            description='Builds a debug version of the libraries')
-
+    variant('debug', default=False, description='Builds a debug version of the libraries')
+    depends_on('cmake@3.12.0:', type='build')
```
Comment (baberlevi) [inline var/spack/repos/builtin/packages/openexr/package.py:25]: I assume you're adding this for the version constraint
I suggest adding a comment pointing to https://github.com/AcademySoftwareFoundation/openexr/blob/master/INSTALL.md#prerequisites

**Review by baberlevi — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/openexr/package.py (hunk 1 above)
Comment (dorton21) [reply var/spack/repos/builtin/packages/openexr/package.py:13]: When a boilerplate Spack package is created it comes with the extra spaces leading to the '=' symbols being aligned. I added the spaces to match that precedence. However, I can revert it to only one space if you think it's appropriate.

**Pull request closed without merging**